### ISSUE TITLE
Stage 1.3: Test root embeddings quality (embedding collapse, semantic clusters)

### ISSUE THREAD

**Issue by marctjones** [marctjones/klareco#619]
# Task

Validate root embeddings quality to ensure no embedding collapse and proper semantic clustering.

## Parent Epic

Part of **#616** (EPIC: Tier-Filtered Model Retraining)

### 1. Function Word Exclusion ✅
```python
checkpoint = torch.load('models/root_embeddings/best_model.pt')
vocab = checkpoint['idx_to_root']
function_words = ['mi', 'kaj', 'la', 'de', 'mal', 'iĝ', 'ig', 'ej', 'en', 'al']
found = [w for w in function_words if w in vocab.values()]
assert len(found) == 0, f"Found function words: {found}"
```

### 2. No Embedding Collapse 🎯
```python
embeddings = checkpoint['model_state_dict']['embeddings.weight']
cos_sim = torch.nn.functional.cosine_similarity(
    embeddings.unsqueeze(0), embeddings.unsqueeze(1), dim=2
)
mean_sim = (cos_sim.sum() - cos_sim.diagonal().sum()) / (cos_sim.numel() - len(cos_sim))
assert mean_sim < 0.5, f"Embedding collapse! Mean similarity: {mean_sim:.4f}"
```

### 3. Semantic Cluster Separation
Test similar/dissimilar word pairs:
```python
similar_pairs = [('hund', 'kat'), ('bon', 'bel'), ('grand', 'vast')]
dissimilar_pairs = [('hund', 'tablo'), ('bon', 'rapid'), ('grand', 'fluid')]
gap = mean_similar - mean_dissimilar
assert gap > 0.03, f"Clusters not separated! Gap: {gap:.4f}"
```

### 4. Vocabulary Coverage
```python
# All Fundamento roots present
fundamento_roots = load_fundamento_roots()  # 2,153 roots
missing = [r for r in fundamento_roots if r not in vocab.values()]
assert len(missing) == 0, f"Missing Fundamento roots: {missing[:10]}"
```

## Acceptance Criteria

- [ ] ✅ No function words in vocabulary
- [ ] 🎯 Mean pairwise similarity < 0.5 (no collapse)
- [ ] 📊 Semantic cluster gap > 0.03
- [ ] 📚 100% Fundamento coverage (tier 1a+1b)
- [ ] 📈 Similar word pairs: similarity > 0.6
- [ ] 📉 Dissimilar word pairs: similarity < 0.3

## Test Script

```bash
python scripts/validate_tier_filtered_training.py
python -m pytest tests/test_embedding_quality.py -v
```

## Dependencies

- Blocked by: #618 (Train root embeddings)
- Blocks: #620 (Integration)

**Comment by marctjones** [comment]
Closing as part of a 2026-09-08 backlog cleanup (see PLAN.md, DESIGN.md, CLAUDE.md's merge gate).

This issue predates the merge gate (adopted 2026-07-13) and the AST-native orchestrator rewrite. It describes work against the pre-orchestrator RAG pipeline or the learned model stack (M1/M2 embeddings, compositional embeddings, entity classifiers, neural rerankers) that was pruned from the repo -- klareco/embeddings/, klareco/models/, and klareco/summarization/ no longer exist. It has no milestone and was never benchmarked, matching exactly the failure mode CLAUDE.md documents (25 capability issues filed 2026-05-26, 8 implemented, none benchmarked).

Labeled archived-2026-09 so this batch is recoverable: gh issue list --state closed --label archived-2026-09. If this capability is still wanted, re-file it against the current architecture naming the benchmark metric and test the merge gate requires.
